index.js

```diff
@@ -648,6 +648,16 @@ window.addEventListener("load", function() {
       } else {
         displayOutcome("motion", "error")("Device Motion is not supported");
       }
+    },
+    "window-close":function(){
+      if(confirm("Confirm before Closing Window?")){
+        addEventListener('beforeunload',function(event){
```

Comment:
Nit: Although `globalThis == window` by default, I'd suggest specifying it.
```suggestion
        window.addEventListener('beforeunload',function(event){
```